src/components/Calculator.js

```diff
@@ -1,77 +1,68 @@
-import React from 'react';
-import './Calculator.css'; // Import the CSS for styling
+import React, { useState } from 'react';
+import './Calculator.css';
+import calculate from '../logic/calculate';
 
 const Calculator = () => {
+  const [calculatorData, setCalculatorData] = useState({});
+  const handleButtonClick = (buttonName) => {
+    console.log('Button clicked:', buttonName); // Add this line
+    if (buttonName === '=') {
+      const result = calculate(calculatorData, '=');
+      setCalculatorData(result);
+    } else {
+      const newData = calculate(calculatorData, buttonName);
+      console.log('New data:', newData);
+      setCalculatorData(newData);
+    }
+  };
+
+  const renderButton = (label, ClassName) => {
+    return (
+      <button
+        type="button"
+        className={ClassName}
+        onClick={() => handleButtonClick(label)}
+      >
+        {label}
+      </button>
+    );
+  };
+
   return (
     <div className="calculator">
-      <div className="display">0</div>
+      <div className="display">
+        {' '}
+        {calculatorData.total || calculatorData.next || '0'}
+      </div>
```

Comment:
- You have done great so far :+1: :+1: Implementing events, but there are a few changes you need to make before moving to the next milestone. :smile: :small_airplane:  

- [ ] Looking at the video below, you can see that when I enter a number, it shows on the screen, but after selecting an operator, the numbers that follow are not shown on the screen. this is not proper as the user is meant to see what he/she is entering on the screen. I'll suggest you update the display logic. :+1: :smile: :wave: 

- - Video reference:

https://github.com/Kaiserabbas/math-magician/assets/107632780/71864a6f-4ef0-47ee-b53f-7d42e898688a

Reply:
I have updated the logic that after selecting an operator, the numbers that follow are displayed properly.